File: frontend/src/types/sprint.ts
export enum SprintStatus {
  PLANNED = 'PLANNED',
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED',
}

export interface Sprint {
  id: number;
  name: string;
  startDate?: string;
  endDate?: string;
  status: SprintStatus;
  goalText?: string;
  capacity?: number;
  boardId: number;
  cardCount?: number;
  totalPoints?: number;
  completedPoints?: number;
}

export interface SprintCreateRequest {
  name: string;
  startDate?: string;
  endDate?: string;
  goalText?: string;
  capacity?: number;
  boardId: number;
}

export interface SprintSnapshot {
  id: number;
  sprintId: number;
  snapshotDate: string;
  totalPoints: number;
  completedPoints: number;
  remainingPoints: number;
  statusCounts: Record<string, number>;
}

File: backend/src/main/java/com/kanban/sprint/Sprint.java
package com.kanban.sprint;

import java.time.LocalDate;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.kanban.board.Board;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "sprints")
@Getter
@Setter
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Sprint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SprintStatus status = SprintStatus.PLANNED;

    @Column(name = "goal_text", columnDefinition = "TEXT")
    private String goalText;

    private Integer capacity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_id", nullable = false)
    @JsonIgnore
    private Board board;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
